### ISSUE TITLE
Boolean Flags for TOPP tools are not working

### ISSUE THREAD

**Issue by t0mdavid-m** [OpenMS/streamlit-template#90]
If boolean parameters are adjusted within the TOPP tool framework, the framework attempts to set the flags explicitly, which is not expected by TOPP tools and leads to errors:

`true` values:
Current Output: `-myflag true`
Expected by TOPP: `-myflag`

`false` values:
Current Output: `myflag false`
Expected by TOPP: (omission)

It seems like we cannot distinguish boolean values from string values from the respective `ini` files (#59).  Thus, for FLASHViewer we went with rules for the parameters values `"true"` and `"false"` (see d1a85b9). However, this solution could cause issues in scenarios where the values `"true"` or `"false"` should be actual strings.

I would assume that this a rare case but also wanted to see if anyone else has another opinion?

**Comment by cbielow** [comment]
Ideas:
1) create an INI file (which uses `true/false` internally) with your parameters and pass that on the commandline
2) inspect the INI file, to determine the value type (a flag will have `<ITEM name="..." type="bool" ....>`
3) just annotate params in streamlit 'manually' as what they are and use that annotation when invoking the TOPP tool (least stable nor fun solution).

**Comment by t0mdavid-m** [comment]
@t0mdavid-m will look into if that is still current.

**Comment by jcharkow** [comment]
Guidance:
In the Ini File Parser (https://github.com/OpenMS/streamlit-template/blob/main/src/workflow/ParameterManager.py) add an option that if the type is bool than to use a true/false flag.

**Comment by jcharkow** [comment]
@t0mdavid-m  look into this and see if it is fixed

**Comment by t0mdavid-m** [comment]
It is not fixed. One good example is the advanced parameter `masstrace_snr_filtering` in the `FeatureFinderMetabo` TOPP tool (TOPP Workflow Framework -> Configure -> Show Advanced Parameter). If it is set to `True`, the workflow will crash as flags are implicit.

I am happy to provide further support if some from GSOC wants take up this issue!

**Comment by subCode321** [comment]
I would like to work on this @t0mdavid-m

**Comment by Imama-Kainat** [comment]
Hello! would be hapyy to work on it

**Comment by Imama-Kainat** [comment]
Hi @t0mdavid-m 

I have analyzed the issue with boolean flags in the TOPP tools framework  and identified the root cause. Here’s my understanding of the problem and the solution I’m considering:

 **Problem Summary:**
Boolean parameters in INI files are stored as strings ("true" / "false") instead of actual booleans.
TOPP tools expect different formats for boolean flags:
Some tools (e.g., FeatureFinderMetabo) expect implicit flags (-flag for True, omit for False).
Some tools expect explicit values (-flag true / -flag false).
The current implementation always sets explicit values, causing errors when implicit flags are needed.


 **Proposed Solution:**
Modify CLI Command Generation to handle boolean flags correctly.
Introduce a Tool-Specific Handling Function to check how each tool expects boolean parameters.
Apply the Correct Flag Formatting Based on the Tool:
If the tool expects implicit flags → -flag for True, omit for False.
Otherwise → -flag true / -flag false.

**Comment by Imama-Kainat** [comment]
Am i  on the right track before proceeding with implementation?

**Comment by t0mdavid-m** [comment]
Yes looks like you are on the right track. Just note that there are both `string` and `bool` parameters with the values `true` and `false`. `string` parameters should be handled as explicit flags, while `bool` parameters should be handled as implicit flags. 

You can distinguish them in the `.ini` files:
`<ITEM name="..." type="bool" ....>`
`<ITEM name="..." type="string" ....>`

**Comment by t0mdavid-m** [comment]
@cbielow suggested a few solutions further up. I think it would be most elegant to inspect the `.ini` file when loading the parameters.
